### PR TITLE
finish - CardTray component and "edit-card" feature

### PR DESCRIPTION
**Pull request by usamavarikkottil**
## Changes
- [x] New `CardTray` component.
- [x] `CardTray` component opens upon clicking a card.
- [x] Ability to edit cards through the `CardTray` component.
- [x] Ability to delete cards through the `CardTray` component.

### CODE REVIEW

**Comment by sz3ry** [comment]
just resolved local conflicts

File: src/actions/card.js [hunk 1]
```diff
@@ -8,11 +8,30 @@ export function addCard(event) {
     }))
 }
 
-export function deleteCard(columnId, card) {
+export function deleteCard(columnId, cardId) {
     board.update(columns => columns.map(column => {
         if(column.id == columnId) {
-            column.cards = column.cards.filter(c => c.id != card.id);
+            column.cards = column.cards.filter(c => c.id != cardId);
```
Comment (ChandlerBaskins) [inline src/actions/card.js]: prefer strict checks !==

File: src/components/card.svelte [hunk 2]
```diff
@@ -9,12 +9,14 @@ import { draggable } from '../actions/draggable';
   function select() {
     dispatch('taskSelected', {
       column: column,
-      task: task,
+      card: card,
```
Comment (ChandlerBaskins) [inline src/components/card.svelte]: you can use  es6 short hand here for better  conciseness. IE {card}

File: src/components/cardTray.svelte [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  const dispatch = createEventDispatcher()
+
+  function save() {
+    if (!cardTitle) {
+      alert('A card must have a title.')
+      return
+    }
+
+    dispatch('cardUpdated', {
+      column: column,
```
Comment (ChandlerBaskins) [inline src/components/cardTray.svelte]: we can use es6 shorthand here for more conciseness {column}

File: src/components/cardTray.svelte [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    <input
+      id="title"
+      bind:value={cardTitle}
+      class="p-2 rounded bg-gray-200 text-gray-800	"
+    />
+  </div>
+  <!-- Change columns of cards by choosing column from a dropdown -->
+  <!-- <div class="flex flex-col space-y-1">
+    <label for="status" class="text-sm">Status</label>
+    <select id="status" bind:value={column} class="p-2 rounded">
```
Comment (ChandlerBaskins) [inline src/components/cardTray.svelte]: Should remove commented out code

File: src/components/column.svelte [hunk 5]
```diff
@@ -105,11 +105,11 @@
...
     {#each cards as card}
       <div class="flex flex-row">
-        <Card column={id} {card} />
+        <Card on:taskSelected column={id} {card} />
         <button
           class="text-white text-xs	 bg-red-500"
-          on:click={deleteCard(id, card)}
-        >
+          on:click={deleteCard(id, card.id)}
+          ><!-- deleteCard(columnID, cardID) -->
```
Comment (ChandlerBaskins) [inline src/components/column.svelte]: should remove commented out code

**Review by ChandlerBaskins — CHANGES_REQUESTED** [review]

**Review by brittneypostma — APPROVED** [review]

**Review by brittneypostma — APPROVED** [review]

**Pull request merged**